subsys/usb/device_next/class/usbd_cdc_acm.c

```diff
@@ -0,0 +1,1066 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/sys/ring_buffer.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/usb/usbd.h>
+#include <zephyr/usb/usb_ch9.h>
+#include <zephyr/usb/class/usb_cdc.h>
+
+#include <zephyr/drivers/usb/udc.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(usbd_cdc_acm, CONFIG_USBD_CDC_ACM_LOG_LEVEL);
+
+/* FIXME: buffer count per device */
+NET_BUF_POOL_FIXED_DEFINE(cdc_acm_ep_pool,
+			  2, 512,
+			  sizeof(struct udc_buf_info), NULL);
+
+#define CDC_ACM_DEFAULT_LINECODING	{sys_cpu_to_le32(115200), 0, 0, 8}
+#define CDC_ACM_DEFAULT_BULK_EP_MPS	0
+#define CDC_ACM_DEFAULT_INT_EP_MPS	16
+#define CDC_ACM_DEFAULT_INT_INTERVAL	0x0A
+
+#define CDC_ACM_CLASS_ENABLED		0
+#define CDC_ACM_CLASS_SUSPENDED		1
+#define CDC_ACM_IRQ_RX_ENABLED		2
+#define CDC_ACM_IRQ_TX_ENABLED		3
+#define CDC_ACM_RX_FIFO_BUSY		4
+#define CDC_ACM_TX_FIFO_BUSY		5
+
+#define CONFIG_USBD_CDC_ACM_STACK_SIZE	512
```

Comment:
This stack is too low for default configuration builds. The "only first packet works" issues are gone when I increase the stack to 1024.

The build command I use is:
`OVERLAY_CONFIG=usbd_next_prj.conf west build --pristine -b nrf52840dk_nrf52840 -d build-cdc-acm-next ~/zephyrproject/zephyr/samples/subsys/usb/cdc_acm`